atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/watch/LockWatchEventCacheIntegrationTest.java

```diff
@@ -273,15 +273,18 @@ public void largerUpdateAfterSmallUpdateOnlyPicksUpNewEvents() {
     }
 
     @Test
-    public void getCommitUpdateIsInvalidatedAllIfEventsHaveBeenDeleted() {
+    public void getCommitUpdateIsInvalidateSomeAsEventsAreRetained() {
         createEventCache(2);
         setupInitialState();
         eventCache.processGetCommitTimestampsUpdate(COMMIT_UPDATE, SUCCESS);
         eventCache.processStartTransactionsUpdate(ImmutableSet.of(), SUCCESS_2);
         verifyStage();
 
         CommitUpdate commitUpdate = eventCache.getCommitUpdate(START_TS_1);
-        assertThat(commitUpdate.accept(new InvalidatedAllVisitor())).isTrue();
+        assertThat(commitUpdate.accept(new CommitUpdateVisitor())).containsExactlyInAnyOrder(DESCRIPTOR, DESCRIPTOR_3);
```

Comment:
What is the semantics of this commit update? I can see these are the descriptors that were locked while we were in the commit stage, so we will use this information to invalidate existing cached entries for those?

Reply:
The commit update essentially tells you what events we have seen between the version at start and commit time. This allows you to work out which locks have been taken out during your transaction and thus, if you are a serialisable transaction and you have read a cached value, you can determine whether you need to conflict or not (also, you wrote the commit update originally :P).